ta/pkcs11/src/processing.c

```diff
@@ -422,19 +482,27 @@ enum pkcs11_rc entry_processing_step(struct pkcs11_client *client,
 	if (processing_is_tee_symm(mecha_type))
 		rc = step_symm_operation(session, function, step,
 					 ptypes, params);
+	else if (processing_is_tee_digest(mecha_type))
+		rc = step_digest_operation(session, function, step, obj,
+					   ptypes, params);
 	else
 		rc = PKCS11_CKR_MECHANISM_INVALID;
 
-	if (rc == PKCS11_CKR_OK && step == PKCS11_FUNC_STEP_UPDATE) {
+	if (rc == PKCS11_CKR_OK && (step == PKCS11_FUNC_STEP_UPDATE ||
+				    step == PKCS11_FUNC_STEP_UPDATE_KEY)) {
 		session->processing->updated = true;
 		DMSG("PKCS11 session%"PRIu32": processing %s %s",
 		     session->handle, id2str_proc(mecha_type),
 		     id2str_function(function));
 	}
 
+	if (step == PKCS11_FUNC_STEP_ONESHOT)
+		session->processing->oneshotted = true;
```

Comment:
This is not required.  At line 512, active processing is released for ONESHOT and FINAL operations. So call to any other Digest operation after C_Digest will result in an error. I  think I did test this when testing C_Sign stuff.